src/main/java/org/elasticsearch/indices/cache/query/IndicesQueryCache.java

```diff
@@ -185,8 +178,14 @@ public boolean canCache(ShardSearchRequest request, SearchContext context) {
         if (hasLength(request.templateSource())) {
             return false;
         }
-        // for now, only enable it for search type count
-        if (context.searchType() != SearchType.COUNT) {
+        // for now, only enable it for requests with not hits
+        if (context.size() != 0) {
+            return false;
+        }
+        // We cannot cache with DFS because results depend not only on the content of the index but also
```

Comment:
There is also an issue with search scripts since all search scripts can read the score of the current document. I'll add a comment.